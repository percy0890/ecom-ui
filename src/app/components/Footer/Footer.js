import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Footer.scss';

export default function Footer(props) {
  const [activeItemPath, setActiveItemPath] = React.useState();

  return (
    <ul className="footer">
      {props.menuItems.filter(menuItem => menuItem.sidebar).map(menuItem => {
        const { name, icon, iconActive, path } = menuItem;
        return (
          <NavLink
            exact
            to={path}
            activeClassName="active"
            isActive={(match, location) => {
              if (!match) {
                return false;
              }
              setActiveItemPath(location.pathname);
              return true;
            }}
            key={name}
          >
            <li className="footer-item">
              <img
                src={activeItemPath === path ? iconActive : icon}
                alt={name}
                className="menu-icon"
              />
              <div className="menu-text">{name}</div>
            </li>
          </NavLink>
        );
      })}
    </ul>
  );
}

Footer.propTypes = {
  menuItems: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      icon: PropTypes.string,
      path: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
};
