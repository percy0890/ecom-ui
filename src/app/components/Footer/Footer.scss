@import '~app/styles/core.scss';

.footer {
  display: none;
}

@include media(mobile down) {
  .footer {
    border-top-left-radius: 1.5rem;
    border-top-right-radius: 1.5rem;
    background: $app-bg-white;
    box-shadow: $app-shadow-container-flipped;
    padding: 0;
    height: 8rem;
    display: flex;
    align-items: center;
    position: fixed;
    right: 0;
    bottom: 0;
    left: 0;
    margin: 0;
    list-style: none;

    a {
      flex: 1;
      text-align: center;

      &.active .footer-item .menu-text {
        color: $brand-color-primary;
      }

      .footer-item {
        padding: 1rem 0;

        .menu-icon {
          width: 2rem;
          height: 3rem;
        }

        .menu-text {
          padding-top: 0.5rem;
          font-family: $app-font-bold;
          color: $app-text-primary;
        }
      }
    }
  }
}
