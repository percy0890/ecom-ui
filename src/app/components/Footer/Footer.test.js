import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

const setState = jest.fn();
const useStateSpy = jest.spyOn(React, 'useState');
useStateSpy.mockImplementation(init => [init, setState]);

describe('<Footer />', () => {
  it('renders correctly', () => {
    expect(
      shallow(
        <Footer
          menuItems={[
            {
              exact: true,
              path: null,
              screen: true,
              sidebar: true,
              name: 'Open',
              icon: true,
              iconActive: true,
              isPrivate: true,
            },
          ]}
        />,
      ),
    ).toMatchSnapshot();
  });
});
