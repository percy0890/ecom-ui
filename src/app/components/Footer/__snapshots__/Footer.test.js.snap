// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<ul
  className="footer"
>
  <NavLink
    activeClassName="active"
    exact={true}
    isActive={[Function]}
    key="Open"
    to={null}
  >
    <li
      className="footer-item"
    >
      <img
        alt="Open"
        className="menu-icon"
        src={true}
      />
      <div
        className="menu-text"
      >
        Open
      </div>
    </li>
  </NavLink>
</ul>
`;
